pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -639,17 +639,30 @@ func (c *Cluster) startDeployments(mons []*monConfig, requireAllInQuorum bool) e
 	// 1) New clusters where we are starting one deployment at a time. We only need to check for quorum once when we add a new mon.
 	// 2) Clusters being restored where no mon deployments are running. We need to start all the deployments before checking quorum.
 	onlyCheckQuorumOnce := false
-	deployments, err := c.context.Clientset.AppsV1().Deployments(c.Namespace).List(metav1.ListOptions{LabelSelector: fmt.Sprintf("app=%s", AppName)})
-	if err != nil {
-		if kerrors.IsNotFound(err) {
+	deployments, derr := c.context.Clientset.AppsV1().Deployments(c.Namespace).List(metav1.ListOptions{LabelSelector: fmt.Sprintf("app=%s", AppName)})
+	pods, perr := c.context.Clientset.CoreV1().Pods(c.Namespace).List(metav1.ListOptions{LabelSelector: fmt.Sprintf("app=%s", AppName)})
+
+	if derr != nil || perr != nil {
```

Comment:
Checking for two different conditions at the same time is more difficult for readability and code maintainability. I'd suggest to leave the original code for the deployment, then separately add the new code that tests for the pod existence. It will be very similar code so maybe you can factor out a helper method.

Reply:
Also agreed with @BlaineEXE that we can hopefully check the deployment status instead of looking for pods.

Reply:
yeah makes sense to check for mo's deployment status. Since there is no `status.phase` field for deployments. I think we can check for this condition `deploy.Status.Replicas != 0 && deploy.Status.Replicas != deploy.Status.ReadyReplicas` for marking the deployments as ready.